### PR TITLE
Gaussian network refactor and tests

### PR DESCRIPTION
**Pull request by HenriDeh**
Nothing new compared to #839 It just does not include MPO updates and the content of the other PRs I opened today.

PR Checklist

- [x] Update NEWS.md?
- [x] Unit tests for all structs / functions?
- [x] Integration and correctness tests using a simple env?
- [x] PR Review?
- [x] Add or update documentation?
- [x] Write docstrings for new methods?

### CODE REVIEW

**Comment by HenriDeh** [comment]
This should be it for GausianNetwork. I also remodeled many tests and detected scalar indexing in mvnormlogpdf which I fixed. 
Remaining are the tests for MPO (in another PR). 
@jeremiahpslewis I think we should add a codecov workflow. I'm not certain I covered it all in these tests. Unfortunately, we can't test CUDA stuff with github's workflow. The tests pass on my machine but this will not be visible here and in a codecov report.

**Comment by HenriDeh** [comment]
@jeremiahpslewis I had to add kwarg constructor to Agent to have the RLExperiments tests pass. Do you know why they didn't work here (see two commits ago), but did in your Agent refactor PR ?

**Comment by jeremiahpslewis** [comment]
@HenriDeh honestly, I can't come up with any reasonable explanation for why that might be the case.

**Review by jeremiahpslewis — DISMISSED** [review]

**Comment by jeremiahpslewis** [comment]
@HenriDeh PR looks great!

**Comment by jeremiahpslewis** [comment]
Very much agree about  codecov being useful. It's new to me though. Do you know how to set it up?

> This should be it for GausianNetwork. I also remodeled many tests and detected scalar indexing in mvnormlogpdf which I fixed.
> Remaining are the tests for MPO (in another PR).
> @jeremiahpslewis I think we should add a codecov workflow. I'm not certain I covered it all in these tests. Unfortunately, we can't test CUDA stuff with github's workflow. The tests pass on my machine but this will not be visible here and in a codecov report.

File: src/ReinforcementLearningCore/test/utils/distributions.jl [hunk 1]
```diff
@@ -43,8 +43,8 @@ using Test, LinearAlgebra, Distributions, ReinforcementLearningCore, CUDA
                 σ2 = [6f0 6f0; 5f0 5f0]
                 d2 = MvNormal(μ2[:, 1], LinearAlgebra.Diagonal(map(abs2, σ2[:, 1])))
                 @testset "diagnormlogpdf" begin
+                    logpdfs = diagnormlogpdf(μ1, σ1, x)
                     for i in 1:2
-                        logpdfs = diagnormlogpdf(μ1, σ1, x)
                         @test size(logpdfs) == (1, 2)
```
Comment (jeremiahpslewis) [inline src/ReinforcementLearningCore/test/utils/distributions.jl:48]: This can be puled out of the loop as well, right?

**Review by jeremiahpslewis — CHANGES_REQUESTED** [review]
One minor thing, based on your fix

File: src/ReinforcementLearningCore/test/utils/distributions.jl (hunk 1 above)
Comment (HenriDeh) [reply src/ReinforcementLearningCore/test/utils/distributions.jl:54]: Yes I caught it in the following commit.

**Comment by HenriDeh** [comment]
I still need to restructure CovGaussian tests and fix the CUDA Categorical tests but better do this in a dedicated PR. I caught quite a few issues in this one already, it was for the best.

**Review by jeremiahpslewis — APPROVED** [review]

**Review by jeremiahpslewis — APPROVED** [review]

**Comment by jeremiahpslewis** [comment]
@HenriDeh Looks great!! ok to merge?

**Pull request merged**